gravitee-am-test/specs/gateway/mfa/mfa-general.spec.ts

```diff
@@ -0,0 +1,392 @@
+
+    let xsrf = await extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+    let action = await extractDomAttr(loginPage, 'form', 'action');
+
+    const loginPostResponse = await withRetry(() =>
+      performFormPost(
+        action,
+        '',
+        {
+          'X-XSRF-TOKEN': xsrf,
+          username: testContext.domain.user.username,
+          password: testContext.domain.user.password,
+          rememberMe: 'off',
+          client_id: testContext.application.clientId,
+        },
+        {
+          Cookie: loginPage.headers['set-cookie'],
+          'Content-type': 'application/x-www-form-urlencoded',
+        },
+      ).expect(302),
+    );
+
+    const authResponseFinal = await withRetry(() => followUpGet(loginPostResponse, 302));
+
+    expect(authResponseFinal.headers['location']).toBeDefined();
+    expect(authResponseFinal.headers['location']).toContain(testContext.application.redirectUris[0]);
+    expect(authResponseFinal.headers['location']).toContain('code=');
+  });
+  it('when stepUp is disabled', async () => {
+    const applicationSettings = {
+      ...defaultApplicationSettings(),
+      settings: {
+        mfa: {
+          factor: {
+            defaultFactorId: testContext.domain.factors[0].id,
+            applicationFactors: [testContext.domain.factors[0]],
+          },
+          stepUpAuthenticationRule: '',
+          stepUpAuthentication: { active: false, stepUpAuthenticationRule: '' },
+        },
+      },
+    };
+
+    await patchApplication(testContext.domain.domainId, testContext.admin.accessToken, applicationSettings, testContext.application.id);
+
+    const authResponse = await withRetry(() => get(testContext.oidc.clientAuthorizationEndpoint, 302));
+    const loginPage = await withRetry(() => followUpGet(authResponse, 200));
+
+    let xsrf = await extractDomValue(loginPage, '[name=X-XSRF-TOKEN]');
+    let action = await extractDomAttr(loginPage, 'form', 'action');
+
+    const loginPostResponse = await withRetry(() =>
+      performFormPost(
+        action,
+        '',
+        {
+          'X-XSRF-TOKEN': xsrf,
+          username: testContext.domain.user.username,
+          password: testContext.domain.user.password,
+          rememberMe: 'off',
+          client_id: testContext.application.clientId,
+        },
+        {
+          Cookie: loginPage.headers['set-cookie'],
+          'Content-type': 'application/x-www-form-urlencoded',
+        },
+      ).expect(302),
+    );
+
+    const authResponseFinal = await withRetry(() => followUpGet(loginPostResponse, 302));
+
+    expect(authResponseFinal.headers['location']).toBeDefined();
+    expect(authResponseFinal.headers['location']).toContain(testContext.application.redirectUris[0]);
+    expect(authResponseFinal.headers['location']).toContain('code=');
+  });
+});
+
+describe('With enrolled factor should redirect to Enrollment and Challenge', () => {
+  it('when stepUp is not false', async () => {
```

Comment:
**TODO**: if the purpose of this test is to check that the stepUp prompt the challenge, then this test is not complete. As far as I understand, we only check that the user after login is challenged (because enroll & challenge are active and required) but we never evaluate the stepUp. StepUp is evaluated once the user get a code from the authorize endpoint and request again the authorize endpoint on an active session

Reply:
looks like this comment is still relevant

Reply:
I fixed the test. Now it goes with MFA end to end authorization. Then, on /authorization the Location points to challange due to stepUp=true.